lib/deps/ajax.js

```diff
@@ -129,7 +129,7 @@ function ajax(options, adapterCallback) {
     if (response.statusCode >= 200 && response.statusCode < 300) {
       onSuccess(data, response, callback);
     } else {
-      if (options.binary) {
+      if (options.binary && data.length) {
```

Comment:
Sorry if I am being dumb, why are we `JSON` parsing when we are expecting binary in the first place?